### PR TITLE
[CARBONDATA-2299]Support showing all segment information(include visible and invisible segments)

### PR DESCRIPTION
**Pull request by zzcclp**
Use command 'SHOW HISTORY SEGMENTS' to show all segment information(include visible and invisible segments)

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?  No
 
 - [x] Document update required? 

 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3479/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4706/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4214/

File: integration/spark2/src/test/scala/org/apache/spark/util/CarbonCommandSuite.scala [hunk 1]
```diff
@@ -190,4 +190,21 @@ class CarbonCommandSuite extends Spark2QueryTest with BeforeAndAfterAll {
...
+      + "TBLPROPERTIES('AUTO_LOAD_MERGE'='true','COMPACTION_LEVEL_THRESHOLD'='2,2')")
+    val carbonTable = CarbonMetadata.getInstance().getCarbonTable("default", tableName)
+    sql(s"insert into ${tableName} select 'abc1',1")
+    sql(s"insert into ${tableName} select 'abc2',2")
+    sql(s"insert into ${tableName} select 'abc3',3")
+    assert(sql(s"show segments for table ${tableName}").collect().length == 4)
+    assert(sql(s"show history segments for table ${tableName}").collect().length == 4)
+    sql(s"clean files for table ${tableName}")
+    assert(sql(s"show segments for table ${tableName}").collect().length == 2)
+    assert(sql(s"show history segments for table ${tableName}").collect().length == 4)
```
Comment (jackylk) [inline integration/spark2/src/test/scala/org/apache/spark/util/CarbonCommandSuite.scala]: can you assert the segment name in the collect result

Comment (zzcclp) [reply integration/spark2/src/test/scala/org/apache/spark/util/CarbonCommandSuite.scala]: Done

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3539/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4767/

**Comment by zzcclp** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4769/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3541/

**Comment by zzcclp** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4271/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4770/

**Comment by zzcclp** [comment]
@ravipesala @jackylk please help to review, thanks.

**Comment by zzcclp** [comment]
@ravipesala @jackylk please help to review, thanks.

**Comment by jackylk** [comment]
LGTM

**Pull request closed without merging**